activestorage/test/models/variant_with_record_test.rb

```diff
@@ -57,7 +57,70 @@ class ActiveStorage::VariantWithRecordTest < ActiveSupport::TestCase
     assert_equal "local_public", variant.image.blob.service_name
   end
 
-  test "eager loading" do
+  test "eager loading of single attachment on each record" do
+    def create_users
+      user1 = User.create!(name: "Josh")
+      user2 = User.create!(name: "John")
+
+      blob1 = directly_upload_file_blob(filename: "racecar.jpg")
+      assert_difference -> { ActiveStorage::VariantRecord.count }, +1 do
+        blob1.representation(resize_to_limit: [100, 100]).process
+      end
+
+      blob2 = directly_upload_file_blob(filename: "racecar_rotated.jpg")
+      assert_difference -> { ActiveStorage::VariantRecord.count }, +1 do
+        blob2.representation(resize_to_limit: [100, 100]).process
+      end
+
+      assert_no_difference -> { ActiveStorage::VariantRecord.count } do
+        user1.cover_photo.attach(blob1)
+        user2.cover_photo.attach(blob2)
+      end
+
+      User.where(id: [user1.id, user2.id])
+    end
+
+    users = create_users()
```

Comment:
Can we follow example below and create users only once? 
Reason- The assertions are run twice in the block, we could do reload as needed blow.

Reply:
Thanks for the review, I updated it now. I had some slight reservations about using `reload` because it made the query count assertion slightly unpredictable:

```ruby
users.reload # performs users query
users.each { ...  } # no query performed here

users.reload # performs users query
users.with_attached_cover_photo.each { ... # exact same query is performed again
```

After looking in the source it seems `reset` is what I want - this clears any caches but does not perform the query, so I used that.